libnmstate/nm/ipv6.py

```diff
@@ -46,15 +52,23 @@ def get_info(active_connection):
 
 def create_setting(config):
     setting_ip = nmclient.NM.SettingIP6Config.new()
+    has_valid_addr = False
     if config and config.get('enabled') and config.get('address'):
         setting_ip.props.method = (
             nmclient.NM.SETTING_IP6_CONFIG_METHOD_MANUAL)
         for address in config['address']:
+            addr_str = '%s/%d' % (address['ip'], address['prefix-length'])
+            if is_link_local_addr(addr_str):
+                logging.warning('IPv6 link local address %s is ignored when '
+                                'applying desired state' % addr_str)
+                continue
+            has_valid_addr = True
             naddr = nmclient.NM.IPAddress.new(socket.AF_INET6,
                                               address['ip'],
                                               address['prefix-length'])
             setting_ip.add_address(naddr)
-    else:
+
+    if not has_valid_addr:
```

Comment:
How about looking in the setting_ip  config and check if there are any addresses defined? (instead of using another flag)